### PR TITLE
Add decorate Links

### LINKED ISSUE #400: adobe_mc
Add adobe_mc parameter to all buy links.

### PR DESCRIPTION
**Pull request by andreibogdan**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #400 

Test URLs:
- Before: https://main--bitdefender--hlxsites.hlx.page/solutions/
- After: https://400-adobe_mc--bitdefender--hlxsites.hlx.page/solutions/

### CODE REVIEW

File: solutions/scripts/scripts.js [hunk 1]
```diff
@@ -276,6 +276,30 @@ export default function decorateLinkedPictures(main) {
...
+    .forEach((link) => {
+      // handling links
+      if (link.getAttribute('href').startsWith('https://www.bitdefender.com.au/site/Store/buy')) {
+        // add adobe_mc parameter to the link
+        const url = new URL(link.href);
+        url.searchParams.set('adobe_mc', 'MCAID%3D%7CMCORGID');
+        link.href = url.href;
+      }
+
+      const url = new URL(link.href);
```
Comment (vtsaplin) [inline solutions/scripts/scripts.js:295]: Should we put this above the if and reuse?

File: solutions/scripts/scripts.js [hunk 2]
```diff
@@ -276,6 +276,30 @@ export default function decorateLinkedPictures(main) {
...
+      const external = !url.host.match('macktrucks.com') && !url.host.match('.hlx.(page|live)') && !url.host.match('localhost');
```
Comment (vtsaplin) [inline solutions/scripts/scripts.js:296]: Is `macktrucks.com` related to Bitdefender?

**Review by vtsaplin — CHANGES_REQUESTED** [review]

File: solutions/scripts/scripts.js (hunk 2 above)
Comment (andreibogdan) [reply solutions/scripts/scripts.js:296]: hmmm. good catch :)

File: solutions/scripts/scripts.js [hunk 3]
```diff
@@ -533,3 +573,5 @@ async function loadPage() {
 }
 
 loadPage();
+
+const checkInterval = setInterval(checkAEPDataCollection, 100);
```
Comment (vtsaplin) [inline solutions/scripts/scripts.js:577]: I would add this to `loadLazy`

File: solutions/scripts/scripts.js [hunk 4]
```diff
@@ -295,6 +295,46 @@ export default function decorateLinkedPictures(main) {
...
+  if (window.adobe && window.adobe.target && window.adobe.target.getOffer) {
+    // Your custom code here
+    appendAdobeMcLinks('main');
+    // Stop checking, as AEP Data Collection is now loaded
+    // eslint-disable-next-line  no-use-before-define
+    clearInterval(checkInterval);
+  }
+}
+
+// Set an interval to check every 100 milliseconds (adjust as needed)
```
Comment (vtsaplin) [inline solutions/scripts/scripts.js:336]: It seems like an orphaned comment.

**Review by vtsaplin — CHANGES_REQUESTED** [review]

File: solutions/scripts/scripts.js (hunk 3 above)
Comment (andreibogdan) [reply solutions/scripts/scripts.js]: I added the code to Lazy loading

**Comment by andreibogdan** [comment]
not applicable

**Pull request closed without merging**